### PR TITLE
Move to Objectify from Datastore

### PR DESCRIPTION
**Pull request by lesv**
Please review and comment.

### CODE REVIEW

**Comment by lesv** [comment]
Hold off reviewing -- I want to put more comments in.

**Comment by lesv** [comment]
Ok - Please review @tmatsuo

File: README.md [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+Written by Google Cloud Developer Relations
+
+Please ask questions on [Stackoverflow](http://stackoverflow.com/questions/tagged/google-app-engine)
+
+## Project setup, installation, and configuration
+
+How do I, as a developer, start working on the project?
+
+1. cd final  (or phase1)
+1. `mvn install appengine:devserver`
```
Comment (tmatsuo) [inline README.md]: Why was 'install' necessary?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
+1. What dependencies does it have (where are they expressed) and how do I install them?
+1. Can I see the project working before I change anything?
```
Comment (tmatsuo) [inline README.md]: What are these 2 items for?

File: final/src/main/java/com/example/guestbook/Greeting.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import com.googlecode.objectify.Key;
+
+import java.lang.String;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * The @Entity tells Objectify about our entity.  We also register it in
+ * OfyService.java and OfyHelper.java -- very important. Our primary key @Id is set automatically
+ * by Objectify for us.
```
Comment (tmatsuo) [inline final/src/main/java/com/example/guestbook/Greeting.java]: by Datastore?

File: final/src/main/java/com/example/guestbook/Greeting.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * We add a @Parent to tell the object about its @ancestor. We are doing this to support many
```
Comment (tmatsuo) [inline final/src/main/java/com/example/guestbook/Greeting.java]: "@ancestor"
Remove "@"?

File: final/src/main/java/com/example/guestbook/OfyHelper.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * OfyHelper, a ServletContextListener, is setup in web.xml to run before a JSP is run.  This is
+ * required to let JSP's access Ofy.
+ **/
+public class OfyHelper implements ServletContextListener {
+  public void contextInitialized(ServletContextEvent event) {
+    // This will be invoked as part of a warmup request, or the first user
+    // request if no warmup request was invoked.
+    ObjectifyService.register(Guestbook.class);
```
Comment (tmatsuo) [inline final/src/main/java/com/example/guestbook/OfyHelper.java]: Do you really need to register those classes in 2 files?

File: final/src/main/java/com/example/guestbook/OfyService.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import com.googlecode.objectify.ObjectifyService;
+
+// [START objectify_init]
+/**
+ * We have the OfyService so we can statically register our classes with Objectify - this is
+ * required for normal classes to get initialized.
+ *
+ *
+ */
+public class OfyService {
```
Comment (tmatsuo) [inline final/src/main/java/com/example/guestbook/OfyService.java]: Is this class really needed?

**Comment by tmatsuo** [comment]
Put some comments!

**Comment by lesv** [comment]
+ @halgrimur Please take a look.

**Comment by tmatsuo** [comment]
LGTM, please merge. Can you add an issue for adding some tests?

**Pull request merged**